packages/api/src/external/carequality/command/cq-directory/parse-cq-directory-entry.ts

```diff
@@ -127,10 +135,23 @@ function getCoordinates(address: Address[]): Coordinates | undefined {
   return { lat, lon };
 }
 
-function getState(addresses: Address[] | undefined): string | undefined {
-  if (!addresses) return;
-  if (addresses.length > 0 && addresses[0]?.state) return addresses[0].state.value ?? undefined;
-  return;
+function getAddressFields(addresses: Address[] | undefined): LenientAddress {
+  const address: LenientAddress = {};
+  if (addresses && addresses.length > 0) {
+    const firstAddress = addresses[0];
```

Comment:
What if we built a function to choose the best address?

Could calculate a simple score based on whether it has city, state, address line, and return the address w/ the better score.